File: MetinSifreleme/MetinSifreleme.py
def sifreleme(metin, sifrelem_kodu):
    sifrelenmis_metin = ''
    for char in metin:
        karakter = ord(char) + sifreleme_kodu
        sifrelenmis_metin += chr(karakter)
    print(sifrelenmis_metin)

def sifre_cozme(metin, sifreleme_kodu):
    sifresi_cozulmus_metin = ''
    for char in metin:
        karakter = ord(char) - sifreleme_kodu
        sifresi_cozulmus_metin += chr(karakter)
    print(sifresi_cozulmus_metin)

islem = input("Yapmak istediğiniz işlemi girin : ")
sifreleme_kodu = int(input("Şifreleme kodunu girin : "))
metin = input(islem.capitalize() + " yapılacak metni girin : ")
if islem.lower() == "sifreleme":
    sifreleme(metin, sifreleme_kodu)
elif islem.lower() == "sifre cozme":
    sifre_cozme(metin, sifreleme_kodu)
else:
    print("Ya şifreleme yapabilirsiniz\nYa da şifre çözme yapabilirsiniz")

File: MetinSifreleme/MetinSifreleme.java
package metinsifreleme;

import java.util.Scanner;

public class MetinSifreleme {

    static void sifreleme(String metin, int sifrelemeKodu) {
        String sifrelenmisMetin = "";
        for (int i = 0; i < metin.length(); i++) {
            int karakter = metin.charAt(i) + sifrelemeKodu;
            sifrelenmisMetin += (char) karakter;
        }
        System.out.println(sifrelenmisMetin);
    }

    static void sifreCozme(String metin, int sifrelemeKodu) {
        String sifresiCozulmusMetin = "";
        for (int i = 0; i < metin.length(); i++) {
            int karakter = metin.charAt(i) - sifrelemeKodu;
            sifresiCozulmusMetin += (char) karakter;
        }
        System.out.println(sifresiCozulmusMetin);
    }

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Yapmak istediğiniz işlemi girin : ");
        String islem = reader.nextLine();
        System.out.print("Şifreleme kodunu girin : ");
        int sifrelemeKodu = reader.nextInt();
        reader.nextLine();
        System.out.print('Ş' + islem.substring(1, islem.length()) + " yapılacak metni girin : ");
        String metin = reader.nextLine();
        if (islem.toLowerCase().equals("sifreleme")) {
            sifreleme(metin, sifrelemeKodu);
        } else if (islem.toLowerCase().equals("sifre cozme")) {
            sifreCozme(metin, sifrelemeKodu);
        } else {
            System.out.println("Ya şifreleme yapabilirsiniz\nYa da şifre çözme yapabilirsiniz");
        }
    }
}
